R/source.R

```diff
@@ -13,7 +13,33 @@ source_many <- function(files, encoding = "UTF-8", envir = parent.frame()) {
       source_one(file, encoding, envir = envir),
       error = function(cnd) {
         path <- file.path(basename(dirname(file)), basename(file))
-        msg <- paste0("Failed to load {.file {path}}")
+        is_parse_error <- tryCatch(identical(as.character(cnd$call[1]), "parse"), error = function(e) FALSE)
```

Comment:
```suggestion
        is_parse_error <- is_call(cnd$call, "parse")
```

Reply:
hmm I don't particularly like that this is assuming:

- `parse()` corresponds to `base::parse()` and not some other user-defined function.
- `parse()` failed inside `source()` and not in the user code. Admittedly this would be an odd thing to do, but that's still an ambiguity.

Reply:
I understand, I'm just uncomfortable with this sort of ambiguity in such a low level tool. The new error message does seem better but the old one is not that bad

Reply:
Clickable links would be great though

Reply:
Another problem is that these messages could be translated so the regexp hot-patching is not reliable.

Reply:
Though we can assume the file path is always at the beginning of the message I guess.

Another issue is that we should keep the file name in the parent error as is for other sources of errors, so it's clickable.

By the way replacing the `message` field of foreign conditions is not reliable because there might be (now or in the future) a `conditionMessage()` method.

I pushed a few changes to take care of these concerns.